src/RectangleA.java

```diff
@@ -0,0 +1,263 @@
+    private final int MIN_WIDTH = 1;
+    private final int MIN_HEIGHT = 1;
+
+    // 1. instance variables
+    private int _width;
+    private int _height;
+    private Point _pointSW;
+
+    // 2. constructors
+
+    /**
+     * Construct a new rectangle with the specified width, height. Initialize south-west corner to (0,0)
+     *
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleA(int w, int h) {
+        _width = getProperWidth(w);
+        _height = getProperHeight(h);
+        _pointSW = new Point(0, 0);
+    }
+
+    /**
+     * Construct a new rectangle with the specified width, height and south-west vertex
+     *
+     * @param p - south-western vertex
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleA(Point p, int w, int h) {
+        _width = getProperWidth(w);
+        _height = getProperHeight(h);
+        _pointSW = new Point(p);
+    }
+
+    /**
+     * Construct a new rectangle with the specified vertices
+     *
+     * @param sw - south-western vertex
+     * @param ne - north-eastern vertex
+     */
+    public RectangleA(Point sw, Point ne) {
+        _width = ne.getX() - sw.getX();
+        _height = ne.getY() - sw.getY();
+        _pointSW = new Point(sw);
+    }
+
+    /**
+     * Copy constructor for objects of class RectangleA Constructs a rectangle using another rectangle
+     *
+     * @param r - The rectangle from which to construct the new object
+     */
+    public RectangleA(RectangleA r) {
+        _width = r._width;
+        _height = r._height;
+        _pointSW = new Point(r._pointSW);
+    }
+
+    // 3. getters and setters
+
+    /**
+     * Returns the width of the rectangle
+     *
+     * @return the width of the rectangle
+     */
+    public int getWidth() {
+        return _width;
+    }
+
+    /**
+     * Returns the height of the rectangle
+     *
+     * @return the height of the rectangle
+     */
+    public int getHeight() {
+        return _height;
+    }
+
+    /**
+     * Returns the South-West point of the rectangle
+     *
+     * @return a copy of the s-w point of the rectangle
+     */
+    public Point getPointSW() {
+        return new Point(_pointSW);
+    }
+
+    /**
+     * Sets the width of the rectangle
+     *
+     * @param w - the width of the rectangle to set to
+     */
+    public void setWidth(int w) {
+        if (w >= MIN_WIDTH)
+            _width = w;
+    }
+
+    /**
+     * Sets the height of the rectangle
+     *
+     * @param h - the height of the rectangle to set to
+     */
+    public void setHeight(int h) {
+        if (h >= MIN_HEIGHT)
+            _height = h;
+    }
+
+    /**
+     * Sets the South-West point of the rectangle
+     *
+     * @param p - the S-W point of the rectangle to set to
+     */
+    public void setPointSW(Point p) {
+        _pointSW = new Point(p);
+    }
+
+    // 4. methods
```

Comment:
BTW, why do you set numbers to those comments?

Reply:
This is how Shadi does it in class. I guess he will like checking it this way (:

Reply:
cool.
cool, cool, cool.